src/argilla/server/apis/v0/handlers/users.py

```diff
@@ -54,3 +57,20 @@ def create_user(
     user = accounts.create_user(db, user_create)
 
     return User.from_orm(user)
+
+
+@router.delete("/users/{user_id}", response_model=User, response_model_exclude_none=True)
+def delete_user(
+    *, db: Session = Depends(get_db), user_id: UUID, current_user: User = Security(auth.get_user, scopes=[])
+):
+    user = accounts.get_user_by_id(db, user_id)
+    if not user:
+        # TODO: Forcing here user_id to be an string.
+        # Not casting it is causing a `Object of type UUID is not JSON serializable`.
+        # Possible solution redefining JSONEncoder.default here:
+        # https://github.com/jazzband/django-push-notifications/issues/586
+        raise EntityNotFoundError(name=str(user_id), type=User)
+
+    accounts.delete_user(db, user)
+
+    return User.from_orm(user)
```

Comment:
Here, do you have any problem with UUID?

Reply:
Nope, on Pydantic schemas UUID it's correctly serialized. I'm using it with workspaces without any problem.